core/scale/scale_decoder_stream.hpp

```diff
@@ -155,22 +208,26 @@ namespace kagome::scale {
      */
     template <class T>
     ScaleDecoderStream &operator>>(std::list<T> &v) {
-      v.clear();
+      using mutableT = std::remove_const_t<T>;
+      using size_type = typename std::list<T>::size_type;
+
+      static_assert(std::is_default_constructible_v<mutableT>);
+
       CompactInteger size{0u};
       *this >> size;
 
```

Comment:
It seems we can remove code dublication and unite ```ScaleDecoderStream &operator>>(std::list<T> &v)```
and ```vector``` variants.

Reply:
I haven't found a way to combine the implementation of this without affecting other container types

Reply:
Make addition function with this logic and forward call to it for ```list``` and ```vector``` imls.